### DISCUSSION TITLE
Overlap Metadata Through Various Destinations

### DISCUSSION THREAD

**Discussion by JJWest12** [nextgenhealthcare/connect#5143] [Q&A]
I have a channel that has 3 destinations all of which performing an API call. The API call response is simply a number so in each response transformer I have 

var count = msg.toString()
channelMap.put('count', count)

And I have one metadata column for 'count'. My objective is that each 'count' would be unique to the api calls for each destination however it seems like it is just overlapping each time. Is there a work around this?

**Comment by JJWest12** [comment]
Note:

So essentially I want one metadata column that would display all of the results for the 3 separate API calls

**Reply by jonbartels** [reply]
This might do.
```
var count = msg.toString()
var countMap = channelMap.get('count');
countMap = countMap ? JSON.parse(countMap) : {};
countMap['mydestination'] = count;
channelMap.put('countMap', JSON.stringify(countMap))
```

I use maps because I like them. You could make it an array too.

Another option is to look at each `d#` (`d1`, `d2` and so on) in the responseMap. These contain the response message from each destination. They could be iterated or parsed in a final destination

**Comment by tonygermano** [comment]
Use the connectorMap instead of channelMap.
